src/game/client/draw/drawable.cpp

```diff
@@ -3069,3 +3083,211 @@ void Drawable::Apply_Physics_Xform(Matrix3D *mtx)
+            if (Check_Draw_Bit(DRAW_BIT_DISABLED)) {
+                if (m_tintColorEnvelope == nullptr) {
+                    m_tintColorEnvelope = new TintEnvelope();
+                }
+
+                m_tintColorEnvelope->Play(&s_darkGreyDisabledColor, 30, 30, 0xFFFFFFFE);
+            } else if (Check_Draw_Bit(DRAW_BIT_DAMAGED)) {
+                if (m_tintColorEnvelope == nullptr) {
+                    m_tintColorEnvelope = new TintEnvelope();
+                }
+
+                m_tintColorEnvelope->Play(&s_subdualDamageColor, 150, 150, 0xFFFFFFFE);
+            } else if (Check_Draw_Bit(DRAW_BIT_WEAPON_BONUS)) {
+                if (m_tintColorEnvelope == nullptr) {
+                    m_tintColorEnvelope = new TintEnvelope();
+                }
+
+                if (Is_KindOf(KINDOF_INFANTRY)) {
+                    m_tintColorEnvelope->Play(&s_frenzyColorInfantry, 30, 30, 0xFFFFFFFE);
+                } else {
+                    m_tintColorEnvelope->Play(&s_frenzyColor, 30, 30, 0xFFFFFFFE);
+                }
+            } else {
+                if (m_tintColorEnvelope == nullptr) {
+                    m_tintColorEnvelope = new TintEnvelope();
+                }
+
+                m_tintColorEnvelope->Set_Decay_State();
+            }
+        }
+
+        m_previousDrawBits = m_drawBits;
+
+        if (object != nullptr && !object->Is_Effectively_Dead()) {
+            Clear_Draw_Bit(DRAW_BIT_DEAD);
+        }
+
+        if (m_tintColorEnvelope != nullptr) {
+            m_tintColorEnvelope->Update();
+        }
+
+        if (m_selectionColorEnvelope != nullptr) {
+            m_selectionColorEnvelope->Update();
+        }
+
+        if (m_ambientSound != nullptr) {
+            if (m_ambientSoundEnabled) {
+                if (m_ambientSoundFromScriptEnabled) {
+                    if (!m_ambientSound->m_event.Get_Event_Name().Is_Empty()
+                        && !m_ambientSound->m_event.Is_Currently_Playing()) {
+                        const AudioEventInfo *info = m_ambientSound->m_event.Get_Event_Info();
+
+                        if (info == nullptr && g_theAudio != nullptr) {
+                            g_theAudio->Get_Info_For_Audio_Event(&m_ambientSound->m_event);
+                            info = m_ambientSound->m_event.Get_Event_Info();
+                        }
+
+                        if (info == nullptr || info->Is_Looping()) {
+                            Start_Ambient_Sound(false);
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+void Drawable::Color_Flash(
+    RGBColor const *color, unsigned int attack_frames, unsigned int decay_frames, unsigned int peak_frames)
+{
+    if (m_tintColorEnvelope == nullptr) {
+        m_tintColorEnvelope = new TintEnvelope();
+    }
+
+    if (color) {
+        m_tintColorEnvelope->Play(color, attack_frames, decay_frames, peak_frames);
+    } else {
+        RGBColor c;
+        c.Set_From_Int(0xFFFFFFFF);
```

Comment:
this is a ctor based on GlobalData ctor invoking it

Reply:
Its not, confirmed that.